### DISCUSSION TITLE
WTDTA1 modeling issue

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#268] [Modeling] [answered]
In the Version 1.5.10.post4+g35b822bb, I found an issue with the WTDTA1 model that may need to be fixed.

In WTDTA1Data, an input parameter ``ree`` is used to specify the idx of RenExciter:
```python
        self.ree = IdxParam(mandatory=True,
                            info='Renewable exciter idx',
                            )
```

Then in WTDTA1Model, the `ree` is used as the indexer for `reg`(`ExtParam`). Further, `reg` is used as the indexer for `Sn`(`ExtParam`). However, the trouble is that an indexer should be `IdxParam`.
```python
        self.reg = ExtParam(model='RenExciter', src='reg', indexer=self.ree,
                            export=False,
                            )
        self.Sn = ExtParam(model='RenGen', src='Sn', indexer=self.reg,
                           tex_name='S_n', export=False,
                           )
```

A possible way is to develop a new Param ``ExtIdx``, but that can be overkill for such a rare case. Do you have any better idea to solve it?

Regards,
Jinning

**Comment by cuihantao** [comment]
Not quite clear of the issue. Do you mean that `self.Sn = ExtParam` will reject `self.reg` as an indexer, because `self.reg` is of type `ExtParam`?

I don't think it does type checking at all.

**Reply by jinningwang** [reply]
``ExtParam`` will do `link_ext_param`, which will invoke `_list2array()`. The trouble is that, if the idx is a string, then `ExtParam` will run into an error because the string cannot be converted to float.

**Comment by cuihantao** [comment, marked as answer]
ExtParam takes a vtype argument for the type. You can set it to str, and it should pass.

Regards,
Hantao Cui

**Reply by jinningwang** [reply]
It works, thanks!
